src/encoded/static/components/filegallery.js

```diff
@@ -528,24 +549,61 @@ class RawSequencingTable extends React.Component {
+                            // two accessions -- that's how pairs will sort within a biological
+                            // replicate. Also track the `pairSortId` which is the same as
+                            // `portSortKey` but without the paired_end property concatenated.
+                            partner.pairSortId = file.title < partner.title ? file.title : partner.title;
+                            partner.pairSortKey = `${partner.pairSortId}-${partner.paired_end}`;
+                            file.pairSortId = partner.pairSortId;
+                            file.pairSortKey = `${partner.pairSortId}-${file.paired_end}`;
                             return true;
                         }
                     }
                 }
 
-                // File not part of a pair; add to non-paired list and filter it out
-                nonpairedFiles.push(file);
+                // File not part of a pair; add it to the non-paired list as a copy so we can
+                // mutate them later.
+                nonpairedFiles.push(Object.assign({}, file));
                 return false;
             });
 
+            // Sort the non-paired files by biological replicate, but corresponding index reads
+            // files still need weaving into this array.
+            nonpairedFiles = nonpairedFiles.sort(sortBioReps);
+
+            // Weave index files after each corresponding single-ended raw sequencing file.
+            const nonPairedFilesWithIndex = [];
+            if (indexFiles) {
+                nonpairedFiles.forEach((nonpairedFile) => {
+                    nonPairedFilesWithIndex.push(nonpairedFile);
+                    const matchingIndexFile = indexFiles.find(indexFile => indexFile.index_of.includes(nonpairedFile['@id']));
+                    if (matchingIndexFile) {
+                        nonPairedFilesWithIndex.push(matchingIndexFile);
+                    }
+                });
+                nonpairedFiles = nonPairedFilesWithIndex;
+            }
+
+            // Copy indexFiles to have `pairSortId` and  `pairSortKey` properties similar to the
+            // corresponding reads files’ pairSortKeys. For ease-of-implementation and performance,
+            // we only look for the one file of the pair in `pairedFiles` to get its pairSortKey
+            // root (without appended `paired_end` string) -- not both files.
+            const pairedFileIndexReads = [];
+            if (indexFiles) {
+                indexFiles.forEach((indexFile) => {
+                    const indexFileCopy = Object.assign({}, indexFile);
+                    const matchingPairFile = pairedFiles.find(pairedFile => indexFile.index_of.includes(pairedFile['@id']));
+                    if (matchingPairFile) {
+                        indexFileCopy.pairSortId = matchingPairFile.pairSortId;
+                        indexFileCopy.pairSortKey = `${matchingPairFile.pairSortId}-I`;
```

Comment:
Is the 'I' at the end intentional?

Reply:
Very much, so that it always sorts behind {accession}-1 and {accession}-2. I never explain that though, and I should.